lib/shakespeare_analyzer.rb

```
@@ -0,0 +1,73 @@
+require 'pp'
+require 'nokogiri'
+require 'net/http'
+require 'uri'
+
+class ShakespeareAnalyzer
+  def initialize(file=nil)
+    @file = file
+    if FileTest.exist?(@file) then
+      ## Processing a local file
+      if File.size(@file) == 0 then
+        puts "Empty input file; terminating"
```

Your code seems a little paranoid. It's constantly checking for error states. I think it makes it harder to follow the normal course of your program.

I think readability would be improved if you took a more "let it crash if something weird is going on" approach.